### DISCUSSION TITLE
Regularization failing

### DISCUSSION THREAD

**Discussion by ccis88** [neperfepx/neper#718] [Q&A] [answered]
I have a line of instructions and results, but don't know what the problem is.

neper -T -n 20000 -dim 2 -domain "circle(18):translate(-9,-9)" -morphooptiini "coo:file(cooo),weight:0" -transform "cut(cylinder(0,0,0,0,0,1,8.5),hspace(0,0,-1,0),hspace(0,-1,0,0))"  -morpho voronoi -regularization 1
![image](https://github.com/neperfepx/neper/assets/122538253/12375b47-6695-4cd4-a60d-7a67d035ee09)
Thanks for the answer！！！

**Comment by rquey** [comment]
Please provide all inputs so that we can reproduce the bug.  If you have a smaller example, it would help too.

**Reply by ccis88** [reply]
[cooo.xlsx](https://github.com/neperfepx/neper/files/11977141/cooo.xlsx)
Cooo files are seed points that I copied all into excel.

**Reply by rquey** [reply, marked as answer]
Thanks.  I've opened an issue for this bug (#724). It may take some time to fix.

**Comment by Zdmire** [comment]
@rquey 
Hi, i tried to make regularization, but it shown bug. Could you help me?

 neper -T -n 4200 -domain "circle(100)" -morpho "diameq:0.48*lognormal(1.7,0.4)+0.432*lognormal(2.28,0.05)+0.088*lognormal(1.86,0.45),1-circularity:lognormal(0.1,0.03)" -group mode -morphooptistop "val=1e-2" -reg 1 -dim 2 -format tess,ori,geo,ply.svg -o DNBD

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.10.1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 6 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2024, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/tubvgeo-022/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 4200 -domain circle(100) -morpho
         diameq:0.48*lognormal(1.7,0.4)+0.432*lognormal(2.28,0.05)+0.088*lognormal(1.86,0.45),1-circularity:lognormal(0.1,0.03)
         -group mode -morphooptistop val=1e-2 -reg 1 -dim 2 -format
         tess,ori,geo,ply.svg -o DNBD
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =2.073941038
Info   :     > Iteration  79237: fmin=0.009998254 f=0.009998254
Info   :     > Final solution  : f   =0.009998254 (79237 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Regularizing tessellation...
Info   :   - loop 1/2:  77% del=500Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

Aborted (core dumped)
